common/src/services/crypto.service.ts

```diff
@@ -828,14 +806,16 @@ export class CryptoService implements CryptoServiceAbstraction {
         return await this.getKey();
     }
 
-    private resolveLegacyKey(encType: EncryptionType, key: SymmetricCryptoKey): SymmetricCryptoKey {
+    private async resolveLegacyKey(encType: EncryptionType, key: SymmetricCryptoKey): Promise<SymmetricCryptoKey> {
         if (encType === EncryptionType.AesCbc128_HmacSha256_B64 &&
             key.encType === EncryptionType.AesCbc256_B64) {
             // Old encrypt-then-mac scheme, make a new key
-            if (this.legacyEtmKey == null) {
-                this.legacyEtmKey = new SymmetricCryptoKey(key.key, EncryptionType.AesCbc128_HmacSha256_B64);
+            let legacyKey = await this.activeAccountService.get<SymmetricCryptoKey>(StorageKey.LegacyEtmKey, { skipDisk: true });
+            if (legacyKey == null) {
+                legacyKey = new SymmetricCryptoKey(key.key, EncryptionType.AesCbc128_HmacSha256_B64);
+                await this.activeAccountService.save(StorageKey.LegacyEtmKey, legacyKey, { skipDisk: true });
```

Comment:
I don't think this is changing behavior at all. It's still just checking for a cached legacy key, creating a new one, and saving it to the cache instead. The only change here should be that the cached value is handled by the ActiveAccontService instead of the CryptoService.

Reply:
Ah right, I misunderstood how the function worked. Re-read it and you are correct.